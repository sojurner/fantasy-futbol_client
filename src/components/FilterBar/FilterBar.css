@font-face {
  font-family: noto;
  src: url("../../fonts/NotoSansKR-Regular.otf"); }

.filter-bar {
  grid-column: 2 / 3;
  grid-row: 1/2;
  justify-self: center; }

.club-search,
.name-search {
  height: 1.8rem;
  width: 13rem; }

input::placeholder {
  font-size: 1rem;
  padding: 3px 0 0 5px; }

select {
  margin: 5rem 0;
  font-size: 1.3rem;
  height: 2rem; }

form {
  margin: 0 auto; }

.suggest-photo,
.suggest-name {
  display: inline-block;
  font-family: noto; }

.suggest-name {
  text-align: center;
  font-weight: 700;
  font-family: noto-thin;
  font-size: 1rem;
  padding: 0 0 0.3rem 0.5rem; }

.suggestion-row {
  cursor: pointer;
  color: #585656; }

.suggestion-row:hover {
  background-color: silver;
  color: #ffffff; }

.suggestion-box {
  display: flex;
  flex-direction: column; }
