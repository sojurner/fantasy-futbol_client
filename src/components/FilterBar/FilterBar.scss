@font-face {
  font-family: noto;
  src: url('../../fonts/NotoSansKR-Regular.otf');
}

.filter-bar {
  grid-column: 2 / 3;
  grid-row: 1/2;
  justify-self: center;
}

.club-search,
.name-search {
  height: 2.3rem;
  width: 13rem;
  font-size: 1.3rem;
}

.name-search:focus {
  outline: 0;
}

input::placeholder {
  font-size: 1rem;
  padding: 3px 0 0 5px;
}

select {
  font-size: 1.3rem;
  height: 2rem;
}

form {
  margin: 0 auto;
}

.suggest-photo,
.suggest-name {
  display: inline-block;
  font-family: noto;
}

.suggest-name {
  text-align: left;
  font-weight: 700;
  font-family: noto-thin;
  font-size: 1rem;
  padding: 0 0 0.3rem 0.5rem;
  height: 100%;
  width: 100%;
}

.suggestion-row {
  cursor: pointer;
  font-size: 1rem;
  color: #585656;
  background-color: white;
}

.suggestion-row:hover {
  background-color: rgb(192, 192, 192);
  color: #ffffff;
}

.suggestion-box {
  display: flex;
  flex-direction: column;
  width: 13.3rem;
  position: absolute;
}

.filter-bar {
  font-family: noto;
}

.suggest-club {
  font-size: 1rem;
}
