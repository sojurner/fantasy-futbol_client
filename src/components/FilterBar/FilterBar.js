import React, { Component } from 'react';

import { FilterDropDown } from '../FilterDropDown/FilterDropDown';
import { SearchInput } from '../SearchInput/SearchInput';
import { PlayerSuggestions } from '../PlayerSuggestions/PlayerSuggestions';
import { getSuggestions } from '../../helpers/autoSuggest/autoSuggest';
import * as fetch from '../../helpers/apiCalls/apiCalls';

import './FilterBar.css';

class FilterBar extends Component {
  constructor() {
    super();
    this.state = {
      selectedFilter: '',
      searchInput: '',
      suggestions: null
    };
    }
  });

  const clubSuggestionList = clubSuggestions.map(suggestion => {
    if (suggestion) {
      return (
        <span
          className="suggestion-row"
          onClick={event => searchPlayers(event, 'club')}
        >
          <p className="suggest-club">{suggestion}</p>
        </span>
      );
    }
  });
  return (
    <div className="filter-bar">
      <h3 className="filter-title">Filter Players</h3>
      <select onChange={event => filterPlayersByCountry(event.target.value)}>
        {countries}
      </select>
      <form>
        <input
          className="name-search"
          name="searchedName"
          value={currentSearchName}
          placeholder="Search by name"
          onChange={handleChange}
        />
      </form>
      <div className="suggestion-box">{playerSuggestionList}</div>
      <form>
        <input
          className="club-search"
          onChange={handleChange}
          name="searchedClub"
          value={currentSearchClub}
          placeholder="Search by club"
        />
      </form>
      <div className="suggestion-box">{clubSuggestionList}</div>
    </div>
  );
};
