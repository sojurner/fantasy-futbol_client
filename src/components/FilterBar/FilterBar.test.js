import React from 'react';
import { FilterBar } from './FilterBar';

describe('FilterBar', () => {
  let wrapper;
  beforeEach(() => {
    let filterPlayers = jest.fn();
    let mockSearch = jest.fn();
    let mockHandle = jest.fn();
    let mockCountries = [{ name: 'Argentina', id: '1' }];
    let mockPlayerSuggestions = [
      {
        Name: 'L. Messi',
        Photo: 'https://cdn.sofifa.org/players/4/18/158023.png'
      },
      {
        Name: 'N. Messiniti',
        Photo: 'https://cdn.sofifa.org/players/4/18/236341.png'
      }
    ];
    let mockClubSuggestions = ['Chelsea'];

    wrapper = shallow(
      <FilterBar
        filterPlayersByCountry={filterPlayers}
        searchPlayers={mockSearch}
        handleChange={mockHandle}
        currentSearchClub={'chelsea'}
        currentSearchName={'messi'}
        countries={mockCountries}
        playerSuggestions={mockPlayerSuggestions}
        clubSuggestions={mockClubSuggestions}
      />
    );
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
