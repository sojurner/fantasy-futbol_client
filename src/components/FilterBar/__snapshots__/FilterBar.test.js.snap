// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterBar should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    clubSuggestions={
        Array [
            "Chelsea",
          ]
    }
    countries={
        Array [
            Object {
              "id": "1",
              "name": "Argentina",
            },
          ]
    }
    currentSearchClub="chelsea"
    currentSearchName="messi"
    filterPlayersByCountry={[Function]}
    handleChange={[Function]}
    playerSuggestions={
        Array [
            Object {
              "Name": "L. Messi",
              "Photo": "https://cdn.sofifa.org/players/4/18/158023.png",
            },
            Object {
              "Name": "N. Messiniti",
              "Photo": "https://cdn.sofifa.org/players/4/18/236341.png",
            },
          ]
    }
    searchPlayers={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <select
          onChange={[Function]}
>
          Object {
                    "id": "1",
                    "name": "Argentina",
                  }
</select>,
        <form>
          <input
                    className="name-search"
                    name="searchedName"
                    onChange={[Function]}
                    placeholder="Search by name"
                    value="messi"
          />
</form>,
        <div
          className="suggestion-box"
>
          <span
                    className="suggestion-row"
                    onClick={[Function]}
          >
                    <img
                              className="suggest-photo"
                              height="30"
                              src="https://cdn.sofifa.org/players/4/18/158023.png"
                    />
                     
                    <p
                              className="suggest-name"
                    >
                              L. Messi
                    </p>
          </span>
          <span
                    className="suggestion-row"
                    onClick={[Function]}
          >
                    <img
                              className="suggest-photo"
                              height="30"
                              src="https://cdn.sofifa.org/players/4/18/236341.png"
                    />
                     
                    <p
                              className="suggest-name"
                    >
                              N. Messiniti
                    </p>
          </span>
</div>,
        <form>
          <input
                    className="club-search"
                    name="searchedClub"
                    onChange={[Function]}
                    placeholder="Search by club"
                    value="chelsea"
          />
</form>,
        <div
          className="suggestion-box"
>
          <span
                    className="suggestion-row"
                    onClick={[Function]}
          >
                    <p
                              className="suggest-club"
                    >
                              Chelsea
                    </p>
          </span>
</div>,
      ],
      "className": "filter-bar",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            Object {
              "id": "1",
              "name": "Argentina",
            },
          ],
          "onChange": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "id": "1",
            "name": "Argentina",
          },
        ],
        "type": "select",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <input
            className="name-search"
            name="searchedName"
            onChange={[Function]}
            placeholder="Search by name"
            value="messi"
/>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "name-search",
            "name": "searchedName",
            "onChange": [Function],
            "placeholder": "Search by name",
            "value": "messi",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "type": "form",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="suggestion-row"
              onClick={[Function]}
>
              <img
                            className="suggest-photo"
                            height="30"
                            src="https://cdn.sofifa.org/players/4/18/158023.png"
              />
               
              <p
                            className="suggest-name"
              >
                            L. Messi
              </p>
</span>,
            <span
              className="suggestion-row"
              onClick={[Function]}
>
              <img
                            className="suggest-photo"
                            height="30"
                            src="https://cdn.sofifa.org/players/4/18/236341.png"
              />
               
              <p
                            className="suggest-name"
              >
                            N. Messiniti
              </p>
</span>,
          ],
          "className": "suggestion-box",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <img
                  className="suggest-photo"
                  height="30"
                  src="https://cdn.sofifa.org/players/4/18/158023.png"
/>,
                " ",
                <p
                  className="suggest-name"
>
                  L. Messi
</p>,
              ],
              "className": "suggestion-row",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "suggest-photo",
                  "height": "30",
                  "src": "https://cdn.sofifa.org/players/4/18/158023.png",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              " ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "L. Messi",
                  "className": "suggest-name",
                },
                "ref": null,
                "rendered": "L. Messi",
                "type": "p",
              },
            ],
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <img
                  className="suggest-photo"
                  height="30"
                  src="https://cdn.sofifa.org/players/4/18/236341.png"
/>,
                " ",
                <p
                  className="suggest-name"
>
                  N. Messiniti
</p>,
              ],
              "className": "suggestion-row",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "suggest-photo",
                  "height": "30",
                  "src": "https://cdn.sofifa.org/players/4/18/236341.png",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              " ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "N. Messiniti",
                  "className": "suggest-name",
                },
                "ref": null,
                "rendered": "N. Messiniti",
                "type": "p",
              },
            ],
            "type": "span",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <input
            className="club-search"
            name="searchedClub"
            onChange={[Function]}
            placeholder="Search by club"
            value="chelsea"
/>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "club-search",
            "name": "searchedClub",
            "onChange": [Function],
            "placeholder": "Search by club",
            "value": "chelsea",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "type": "form",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="suggestion-row"
              onClick={[Function]}
>
              <p
                            className="suggest-club"
              >
                            Chelsea
              </p>
</span>,
          ],
          "className": "suggestion-box",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <p
                className="suggest-club"
>
                Chelsea
</p>,
              "className": "suggestion-row",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Chelsea",
                "className": "suggest-club",
              },
              "ref": null,
              "rendered": "Chelsea",
              "type": "p",
            },
            "type": "span",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <select
            onChange={[Function]}
>
            Object {
                        "id": "1",
                        "name": "Argentina",
                      }
</select>,
          <form>
            <input
                        className="name-search"
                        name="searchedName"
                        onChange={[Function]}
                        placeholder="Search by name"
                        value="messi"
            />
</form>,
          <div
            className="suggestion-box"
>
            <span
                        className="suggestion-row"
                        onClick={[Function]}
            >
                        <img
                                    className="suggest-photo"
                                    height="30"
                                    src="https://cdn.sofifa.org/players/4/18/158023.png"
                        />
                         
                        <p
                                    className="suggest-name"
                        >
                                    L. Messi
                        </p>
            </span>
            <span
                        className="suggestion-row"
                        onClick={[Function]}
            >
                        <img
                                    className="suggest-photo"
                                    height="30"
                                    src="https://cdn.sofifa.org/players/4/18/236341.png"
                        />
                         
                        <p
                                    className="suggest-name"
                        >
                                    N. Messiniti
                        </p>
            </span>
</div>,
          <form>
            <input
                        className="club-search"
                        name="searchedClub"
                        onChange={[Function]}
                        placeholder="Search by club"
                        value="chelsea"
            />
</form>,
          <div
            className="suggestion-box"
>
            <span
                        className="suggestion-row"
                        onClick={[Function]}
            >
                        <p
                                    className="suggest-club"
                        >
                                    Chelsea
                        </p>
            </span>
</div>,
        ],
        "className": "filter-bar",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              Object {
                "id": "1",
                "name": "Argentina",
              },
            ],
            "onChange": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "id": "1",
              "name": "Argentina",
            },
          ],
          "type": "select",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <input
              className="name-search"
              name="searchedName"
              onChange={[Function]}
              placeholder="Search by name"
              value="messi"
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "name-search",
              "name": "searchedName",
              "onChange": [Function],
              "placeholder": "Search by name",
              "value": "messi",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="suggestion-row"
                onClick={[Function]}
>
                <img
                                className="suggest-photo"
                                height="30"
                                src="https://cdn.sofifa.org/players/4/18/158023.png"
                />
                 
                <p
                                className="suggest-name"
                >
                                L. Messi
                </p>
</span>,
              <span
                className="suggestion-row"
                onClick={[Function]}
>
                <img
                                className="suggest-photo"
                                height="30"
                                src="https://cdn.sofifa.org/players/4/18/236341.png"
                />
                 
                <p
                                className="suggest-name"
                >
                                N. Messiniti
                </p>
</span>,
            ],
            "className": "suggestion-box",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <img
                    className="suggest-photo"
                    height="30"
                    src="https://cdn.sofifa.org/players/4/18/158023.png"
/>,
                  " ",
                  <p
                    className="suggest-name"
>
                    L. Messi
</p>,
                ],
                "className": "suggestion-row",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "suggest-photo",
                    "height": "30",
                    "src": "https://cdn.sofifa.org/players/4/18/158023.png",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                " ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "L. Messi",
                    "className": "suggest-name",
                  },
                  "ref": null,
                  "rendered": "L. Messi",
                  "type": "p",
                },
              ],
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <img
                    className="suggest-photo"
                    height="30"
                    src="https://cdn.sofifa.org/players/4/18/236341.png"
/>,
                  " ",
                  <p
                    className="suggest-name"
>
                    N. Messiniti
</p>,
                ],
                "className": "suggestion-row",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "suggest-photo",
                    "height": "30",
                    "src": "https://cdn.sofifa.org/players/4/18/236341.png",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                " ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "N. Messiniti",
                    "className": "suggest-name",
                  },
                  "ref": null,
                  "rendered": "N. Messiniti",
                  "type": "p",
                },
              ],
              "type": "span",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <input
              className="club-search"
              name="searchedClub"
              onChange={[Function]}
              placeholder="Search by club"
              value="chelsea"
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "club-search",
              "name": "searchedClub",
              "onChange": [Function],
              "placeholder": "Search by club",
              "value": "chelsea",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="suggestion-row"
                onClick={[Function]}
>
                <p
                                className="suggest-club"
                >
                                Chelsea
                </p>
</span>,
            ],
            "className": "suggestion-box",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <p
                  className="suggest-club"
>
                  Chelsea
</p>,
                "className": "suggestion-row",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Chelsea",
                  "className": "suggest-club",
                },
                "ref": null,
                "rendered": "Chelsea",
                "type": "p",
              },
              "type": "span",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
